File: examples/tipping.R
library(reticulate)
library(tidyverse)
library(patchwork)

## Python dependencies
gc <- import("gym_conservation")
gym <- import ("gym")
sb3 <- import ("stable_baselines3")
## initialize the environment
# see: cost 1, benefit ~ 5
env <- gym$make("conservation-v5", cost = 5, benefit = 150,
                sigma = 0.02, alpha = 0.005)
model = gc$models$user_action(env)

model = sb3$PPO("MlpPolicy", env, verbose=1)
model$learn(total_timesteps=500000)

## Simulate a run with the trained model, visualize result
df = env$simulate(model, reps = 10L)
policy = env$policyfn(model, reps = 10L)


p1 <- df %>% pivot_longer(c(state, action, reward)) %>%
  ggplot(aes(time, value, group=rep)) + geom_line() +
  facet_wrap(~name, ncol=1, scales="free_y")


## Simulate a run with the trained model, visualize result
p2 <- policy %>%
  ggplot(aes(state, action, color=rep)) + geom_point()

p1 / p2

File: examples/tipping.py
import gym
import numpy as np
from stable_baselines3 import A2C, PPO, DDPG, TD3, SAC
from stable_baselines3.common.env_checker import check_env
from stable_baselines3.common.evaluation import evaluate_policy

import gym_conservation


#env = gym.make("conservation-v3", beta = 0.1, sigma=0.01, cost = .5, benefit = 0.5, a = .22)
env = gym.make("conservation-v5")
check_env(env)
model = SAC("MlpPolicy", env, verbose=1, policy_kwargs = dict(net_arch=[256, 256]))
model.learn(total_timesteps=300000)

## Simulate a run with the trained model, visualize result
df = env.simulate(model, reps = 1)
env.plot(df, "results/v5-SAC.png")

## Evaluate model
mean_reward, std_reward = evaluate_policy(model, env, n_eval_episodes=50)
print("v5 with SAC", "mean reward:", mean_reward, "std:", std_reward)
